instrumentation-core/spring/spring-boot-autoconfigure/README.md

```diff
@@ -0,0 +1,428 @@
+
+#### Spring Web MVC Auto Configuration
+
+This feature autoconfigures instrumentation for spring-webmvc controllers by adding a [WebMvcTracingFilter](../spring-webmvc-3.1/src/main/java/io/opentelemetry/instrumentation/springwebmvc/WebMvcTracingFilter.java) bean to the application context. This request filter implements the [OncePerRequestFilter](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/filter/OncePerRequestFilter.html) interface to add OpenTelemetry server spans to incoming requests. [Spring Web MVC - Server Span]('#spring-web-mvc-server-span') show cases a sample span generated by the WebMvcTracingFilter. Check out [opentelemetry-spring-webmvc-3.1](../spring-webmvc-3.1/) to learn more about the OpenTelemetry WebMvcTracingFilter.
+
+#### Spring WebFlux Auto Configuration
+
+Provides autoconfigurations for the OpenTelemtry WebClient ExchangeFilter defined in [opentelemetry-spring-webflux-5.0](../spring-webflux-5.0/). This autoconfiguration instruments all requests sent using Spring's WebClient and WebClient Builder beans using conditional class loaders and bean post processors. This feature is supported for spring webflux versions 5.0+ and can be disabled by adding `opentelemetry.trace.httpclients.enabled=False` to your `resources/applications.properties` file. [Spring Web-Flux - WebClient Span]('#spring-web-flux-webclient-span') show cases a sample span generated by the WebClientFilter. Check out [opentelemetry-spring-webflux-5.0](../spring-webflux-5.0/) to learn more about the OpenTelemetry WebClientFilter.
+
+#### Manual Instrumentation Support - @WithSpan
+
+This feature uses spring-aop to wrap methods annotated with `@WithSpan` in a span.
+
+Note - This annotation can only be applied to bean methods managed by the spring application context. Check out [spring-aop](https://docs.spring.io/spring/docs/3.0.x/spring-framework-reference/html/aop.html) to learn more about aspect weaving in spring.
+
+##### Usage
+
+```java
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.opentelemetry.extensions.auto.annotations.WithSpan;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+/**
+ * Test WithSpan
+ *
+ */
+@Component
+public class TracedClass {
+
+    @Autowired
+    // Tracer bean is provided by
+    // io.opentelemetry.instrumentation.spring.autoconfigure.TracerAutoConfiguration.java
```

Comment:
I think users don't necessarily need to know where in our source code it comes from
```suggestion
    // Tracer bean is provided by spring-boot-autoconfigure
```